Documentation/CodingGuidelines/CglPhp/CodingBestPractices/AccessingTheDatabase.rst

```diff
@@ -1,20 +1,18 @@
-.. include:: /Includes.rst.txt
-.. index:: pair: Coding guidelines; Database
-.. _cgl-database-access:
+..  include:: /Includes.rst.txt
+..  index:: pair: Coding guidelines; Database
+..  _cgl-database-access:
 
 ======================
 Accessing the Database
 ======================
 
-The TYPO3 database should always be accessed using the QueryBuilder of doctrine.
-The :php:`ConnectionPool` class can be used to create
-a :php:`QueryBuilder` instance:
+The TYPO3 database should always be accessed using the :php:`QueryBuilder` of
+Doctrine. The :ref:`ConnectionPool <database-connection-pool>` class should be
```

Comment:
We need a distinction of the Doctrine QueryBuilder by way of Connection Pool and the Extbase Repository way via `$this->createQuery()`.

I believe else people may try to utilize the QueryBuilder within an Extbase Repository too (this chapter says to always do that), which will lead to confusion. Maybe link to https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ExtensionArchitecture/Extbase/Reference/Domain/Repository.html#repository-api here (and also, link back from that page and make a note aboute the QueryBuilder there?!)

Reply:
Oh, true, thanks for catching this, I always tend to forget Extbase

Reply:
This example comes straight from the database chapter. If we change it here, also there the examples should be adapted.

IMHO this has nothing to do with this change, so a merge would be appropriate and then a distinct PR which streamlines all the places and there we can discuss how to handle this. I am looking it you @garvinhicking for a suggestion then ;-)